File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String news="";
        for(char c : s.toCharArray()){
            if(Character.isDigit(c)||Character.isLetter(c))
            news+=c;
        }
        news=news.toLowerCase();
        int l=0,r=news.length()-1;
        while(l<r){
            if(news.charAt(l++)!=news.charAt(r--))return false;
        }
        return true;
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int spt=0;
        int ept=s.size()-1;
        while(spt<ept){
            if(!iswalnum(s[spt]))spt++;
            else if(!iswalnum(s[ept]))ept--;
            else{
                if((isdigit(s[spt]) || isdigit(s[ept]) )&& s[spt]!=s[ept]){
                    return false;
                }
                else if(tolower(s[spt])!=tolower(s[ept])){
                    return false;
                }
                else {
                    spt++;
                    ept--;
                }
            }
        }
        return true;
    }
};